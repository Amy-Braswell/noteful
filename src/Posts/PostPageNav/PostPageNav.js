import React from 'react'
import ApiContext from '../../ApiContext'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import CircleButton from '../../CircleButton/CircleButton'
import { findPost, findPostMember } from '../../posts-helpers'
import './PostPageNav.css'

export default class PostPageNav extends React.Component {
  static defaultProps = {
    history: {
      goBack: () => { 
        window.history.back()
       }
    },
    match: {
      params: {}
    }
  }
  static contextType = ApiContext

  render() {
    const { posts, members, } = this.context
    const { postId } = this.props.match.params
    const post = findPost(posts, postId) || {}
    const member = findPostMember(members, post.memberid)
    return (
      <div className='PostPageNav'>
        <CircleButton
          tag='button'
          role='link'
          onClick={() => this.props.history.goBack()}
          className='PostPageNav__back-button'
        >
          <FontAwesomeIcon icon={['fa', 'chevron-left']} />
          <br />
          Back
        </CircleButton>
        {member && (
          <h3 className='PostPageNav__member-name'>
            {member.name}
          </h3>
        )}
      </div>
    )
  }
}
