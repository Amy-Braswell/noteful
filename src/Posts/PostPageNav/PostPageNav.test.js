import React from 'react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import PostPageNav from './PostPageNav'

it('renders without crashing', () => {

  const div = document.createElement('div')
  ReactDOM.render(<PostPageNav />, div)
  ReactDOM.unmountComponentAtNode(div)
  })

describe(`PostPageNav component`, () => {
  it('renders a .PostPageNav by default', () => {
    const wrapper = renderer
    .create(<PostPageNav />)
    .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it.skip('renders a h3 with member name when in props', () => {
    const props = {
      match: {
        params: {
          postId: 'test-post-id'
        }
      }
    }
    const context = {
      posts: [{ id: 'test-post-id', memberid: 'test-member-id' }],
      members: [{ id: 'test-member-id', name: 'Important' }]
    }

    const h3 = renderer
    .create(<PostPageNav {...props} />, context)
    .find('.PostPageNav__member-name')
    .toJSON()
    expect(h3).toMatchSnapshot()
  })
})