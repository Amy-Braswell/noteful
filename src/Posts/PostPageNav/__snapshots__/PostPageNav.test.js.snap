// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPageNav component renders a .PostPageNav by default 1`] = `
<div
  className="PostPageNav"
>
  <button
    className="NavCircleButton PostPageNav__back-button"
    onClick={[Function]}
    role="link"
  >
    <br />
    Back
  </button>
</div>
`;
